File: 0036-valid-sudoku/0036-valid-sudoku.cpp
class Solution {
public:
    bool isValidSudoku(vector<vector<char>>& board) {
        vector<unordered_set<char>> rows(9), cols(9), boxs(9);
        for(int y = 0; y < 9; y++)
        {
            vector<char> row = board[y];
            for(int x = 0; x < 9; x++)
            {   
                char chr = row[x];
                if( chr != '.')
                {
                    if(rows[x].count(chr) == 0)
                        rows[x].insert(chr);
                    else
                        return false;
                    if(cols[y].count(chr) == 0)
                        cols[y].insert(chr);
                    else
                        return false;
                    int indx = (y / 3) * 3 + x / 3;
                    if(boxs[indx].count(chr) == 0)
                        boxs[indx].insert(chr);
                    else
                        return false;
                }
            }
        }
        return true;
    }
};

File: 0036-valid-sudoku/0036-valid-sudoku.cs
public class Solution {
    public bool IsValidSudoku(char[][] board)
    {
        var cols = new List<HashSet<char>>();
        var rows = new List<HashSet<char>>();
        var boxs = new List<HashSet<char>>();
        for(int i = 0; i < 9; i++)
        {
            cols.Add(new HashSet<char>());
            rows.Add(new HashSet<char>());
            boxs.Add(new HashSet<char>());
        }
        for (int x = 0; x < 9; x++)
        {

            for(int y = 0; y < 9; y++)
            {
                if (board[x][y] != '.')
                {
                    if (cols[y].Contains(board[x][y]))
                    {
                        return false;
                    }
                    if (rows[x].Contains(board[x][y]))
                    {
                        return false;
                    }
                    int index = (x / 3) + 3 * (y / 3);
                    if (boxs[index].Contains(board[x][y]))
                    {
                        return false;
                    }
                    cols[y].Add(board[x][y]);
                    rows[x].Add(board[x][y]);
                    boxs[index].Add(board[x][y]);
                }

            }
        }
        return true;
    }
}